lib/phoenix_live_view/channel.ex

```diff
@@ -237,48 +398,56 @@ defmodule Phoenix.LiveView.Channel do
     :ignore
   end
 
-  defp verified_mount(view, id, parent, assigned_new, session, from, phx_socket) do
+  defp verified_mount(view, id, parent, router, assigned_new, session, params, from, phx_socket) do
     %Phoenix.Socket{endpoint: endpoint} = phx_socket
     Process.monitor(phx_socket.transport_pid)
     parent_assigns = register_with_parent(parent, view, id, assigned_new)
+    %{"uri" => uri, "params" => channel_params} = params
 
     lv_socket =
-      View.build_socket(endpoint, %{
+      View.build_socket(endpoint, router, %{
+        view: view,
         connected?: true,
         parent_pid: parent,
         id: id,
         assigned_new: {parent_assigns, assigned_new}
       })
 
-    case view.mount(session, lv_socket) do
-      {:ok, %Socket{} = lv_socket} ->
-        {diff, new_state} =
-          lv_socket
-          |> View.prune_assigned_new()
-          |> build_state(phx_socket, view)
-          |> render_diff(lv_socket)
+    with {:ok, %Socket{} = lv_socket} <- view.mount(session, lv_socket),
+         state = lv_socket |> View.prune_assigned_new() |> build_state(phx_socket) do
 
-        GenServer.reply(from, {:ok, %{rendered: diff}})
-        {:noreply, new_state}
+      case mount_handle_params(uri, channel_params, state) do
```

Comment:
This should work the same as the other handle_params (i.e. looping, and what not).